File: srxproperty/src/dataObject/ListingDetailPO.js
import { ObjectUtil, PropertyTypeUtil, CommonUtil, StringUtil } from "../utils";
import { ListingPO, AmenitiesCategoryPO } from "../dataObject";

class ListingDetailPO {
  constructor(data) {
    if (!ObjectUtil.isEmpty(data)) {
      this.listingPO = new ListingPO(data.listingPO);

      this.completeDate = data.completeDate;
      this.developer = data.developer;
      this.facilities = data.facilities;
      this.features = data.features;
      this.fixtures = data.fixtures;
      this.fuzzyLatitude = data.fuzzyLatitude;
      this.fuzzyLongitude = data.fuzzyLongitude;
      this.latitude = data.latitude;
      this.longitude = data.longitude;
      this.amenitiesCategories = [];
      if (
        !ObjectUtil.isEmpty(data.amenitiesCategories) &&
        Array.isArray(data.amenitiesCategories)
      ) {
        data.amenitiesCategories.map(item => {
          this.amenitiesCategories.push(new AmenitiesCategoryPO(item));
        });
      }
    }
  }
}

export { ListingDetailPO };

File: agent-connect-android/app/src/main/java/sg/searchhouse/agentconnect/model/api/listing/ListingDetailPO.kt
package sg.searchhouse.agentconnect.model.api.listing

import com.google.gson.annotations.SerializedName
import sg.searchhouse.agentconnect.model.api.common.NameValuePO

data class ListingDetailPO(
    @SerializedName("listingPO")
    val listingPO: ListingPO,
    @SerializedName("developer")
    val developer: String,
    @SerializedName("fuzzyLatitude")
    val fuzzyLatitude: String,
    @SerializedName("fuzzyLongitude")
    val fuzzyLongitude: String,
    @SerializedName("amenitiesCategories")
    val amenitiesCategories: List<AmenitiesCategoryPO>,
    @SerializedName("primaryAmenitiesGroups")
    val primaryAmenitiesGroups: List<AmenitiesGroupPO>,
    @SerializedName("features")
    val features: List<NameValuePO>,
    @SerializedName("fixtures")
    val fixtures: List<NameValuePO>,
    @SerializedName("facilities")
    val facilities: List<NameValuePO>,
    @SerializedName("area")
    val area: List<NameValuePO>
)